### ISSUE TITLE
Kindle Scribe: Power button wakes up when cover closed

### ISSUE THREAD

**Issue by mergen3107** [koreader/koreader#11311]
* KOReader version: v2023.10-82
* Device: Kindle Scribe (5.16.2)

#### Issue
When cover is closed, power button press wakes up Kindle.

#### Steps to reproduce
Attached find a fresh crash.log with these actions:
1) Start KOReader via KUAL
2) It opens with last book.
3) Close cover to put Kindle to sleep.
4) Press button. Kindle Wakes up (I can see light from under the cover.
Expected: keep it sleeping if cover is closed.
5) Open the cover, then closed it again - Kindle goes back to sleep.

Isn't this supposed to apply to Kindle as well?
https://github.com/koreader/koreader/blob/5d2a4410648dfc993e6dbc01cbf23941a92404c3/frontend/device/generic/device.lua#L314-L322

Because:
https://github.com/koreader/koreader/blob/5d2a4410648dfc993e6dbc01cbf23941a92404c3/frontend/device/kindle/device.lua#L150

##### `crash.log`
[power_button_while_sleep_crash1.txt](https://github.com/koreader/koreader/files/13802729/power_button_while_sleep_crash1.txt)

**Comment by mergen3107** [comment]
FWIW, Kindle's own system does the same :D 
So, maybe KOReader won't be able to override this.

**Comment by mergen3107** [comment]
Ah, I see. The function `Device:onPowerEvent(ev)` is only used by Kobo, Cervantes, and reMarkable.

So, there is no way to implement this on Kindle?

**Comment by NiLuJe** [comment]
No, we don't handle PM on Kindle, we leave all of it to powerd. I'd have said if this happens, I suspect you can reproduce this in the native reader, and you just did ;).

**Comment by mergen3107** [comment]
Thanks @NiLuJe !

On the other end then, do you know if there is a timer specifically for when KOReader is woken up at the screensaver (with gesture unlock setup)? 

I don't know what keywords to use to search through files. 

Use case:
- Kindle wakes from accidental power button press in the backpack / cover misaligned (which also causes wake up and happened too many times :D )
- A configurable timer kicks in to put Kindle back to sleep, which is different / shorter than the usual suspend time (currently in Top Menu - Device - Autosuspend timeout, the Auto power save plugin).

**Comment by Frenzie** [comment]
Is the power button on this device really so simple to trigger accidentally? We do what you describe for cover wake events because a bit a shift or an external magnet can fairly easily accidentally trigger it.

**Comment by mergen3107** [comment]
Power button is quite sturdy unless gets bumped to the side. It is located near the middle of the side, where the case also has cutouts on front and back which are supposed to help store the pen :D  
The case looks like this: https://www.amazon.com/Ayotu-Stand-Kindle-Scribe-Released/dp/B0BY21DCKX

**Comment by mergen3107** [comment]
Thanks! Protection from cover's slight misalignment works great! It either doesn't trigger wake up when it shifted slightly, or goes back to sleep very soon after it is shifted further to wake up and put back.

**Comment by NiLuJe** [comment]
> do you know if there is a timer specifically for when KOReader is woken up

=>

> No, we don't handle PM on Kindle, we leave all of it to powerd.

**Comment by NiLuJe** [comment]
Depending on the device (I don't necessarily remember this being always true, which would be a problem in and of itself), powerd *might* tell us why it was woken up:

https://github.com/koreader/koreader/pull/11174#issuecomment-1830064445

The problem is, I don't know how stable this is, I don't have the real, full mapping of those magic numbers, and the input backend doesn't actually forward that (but it technically could, like the evdev stuff does on Kobo).

Also, since we still do not control PM, it would be fairly brittle and hellish to test.

The bit of Generic code quoted above took *a lot* of trial and error to get right.

TL;DR: No. ;p.

**Comment by mergen3107** [comment]
> Top Menu - Device - Autosuspend timeout

OK :D thanks! At least I am happy that extending this timeout works.

**Comment by NiLuJe** [comment]
And what an adventure that tiny little feature was, let me tell you >_<".

**Comment by mergen3107** [comment]
Thank you @NiLuJe !
